### DISCUSSION TITLE
contributions.filter is not a function

### DISCUSSION THREAD

**Discussion by jayvdb** [lowlighter/metrics#1148] [General]
For the notable plugin, I set `debug: true` and `plugin_notable_types: commit, issue, pull_request`, and obviously have `indepth` enabled, and now I am seeing quite a few of the following in the log

```
metrics/compute/jayvdb/plugins > notable > indepth > successfully processed ansible/ansible-lint
TypeError: contributions.filter is not a function
      at Object.default [as notable] (file:///metrics/source/plugins/notable/index.mjs:83:41)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async file:///metrics/source/plugins/core/index.mjs:67:30
      at async Promise.all (index 0)
      at async metrics (file:///metrics/source/app/metrics/index.mjs:64:22)
      at async retry.retries.retries (file:///metrics/source/app/action/index.mjs:398:34)
      at async retry (file:///metrics/source/app/action/index.mjs:57:22)
      at async file:///metrics/source/app/action/index.mjs:397:20
  metrics/compute/jayvdb/plugins > notable > indepth > failed to compute for pdm-project/pdm
  metrics/compute/jayvdb/plugins > notable > indepth > successfully processed TheLartians/ModernCppStarter
```
Run: https://github.com/jayvdb/jayvdb/runs/7468279468?check_suite_focus=true

I don't have any commits in that repo, but I did create an issue: https://github.com/pdm-project/pdm/issues/created_by/jayvdb

pdm now appears in my svg https://github.com/jayvdb/jayvdb/commit/01ec97176e18086d5cd4bd0fc70d35b065ea5698 without any of the numerical scores inside its circle.

So I am guessing that `plugin_notable_types` of `issue` & `pull_request` are incompatible with `indepth` at the moment, unless there was a commit also in the repo.

But the problem is a bit more nuanced than that, as I only have a rejected issue in https://github.com/TheLartians/ModernCppStarter/issues/131, and only an unmerged closed PR in https://github.com/ansible/ansible-lint/pull/2196 - both repos that were "successfully processed".

**Comment by jayvdb** [comment]
Now I look more closely at the issues and pull requests, they are always "1".  Even for "py2many" where I have [99 merged pulls](https://github.com/py2many/py2many/pulls?q=is%3Apr+author%3Ajayvdb+is%3Aclosed).

Also I have added `repositories_forks: true` and `repositories_affiliations: owner, collaborator, organization_member`, and I still cant seem to get "coala" to appear at all, despite [100+ merged PRs](https://github.com/coala/coala/pulls?q=is%3Apr+author%3Ajayvdb+is%3Aclosed), and "pywikibot" where I have [800+ commits](https://github.com/wikimedia/pywikibot/graphs/contributors).

However pywikibot is a less common case where there are no PRs - all dev is done in https://gerrit.wikimedia.org and a bot sync's commits to GitHub, so there wont be a event in my user push event stream, however the normal GitHub profile manages to display these events.  A lot of larger orgs use this type of approach either because they dont like GitHub's workflow, or they don't trust the commercial GitHub as permanent archive and only use it as a secondary backup that is accessible to a wider audience.

**Reply by lowlighter** [reply]
Didn't dig too much into it yet, but I remember if some indepth queries cannot be resolved (like API timeouts) then they're ignored by the plugin, which may be your current issue

Anyway I'll open #1157 so it's eventually investigated 👍
